aiida/orm/querybuilder.py

```diff
@@ -1773,11 +1746,22 @@ def queryhelp(self):
         The queryhelp can be used to create a copy of the QueryBuilder instance like so::
 
             qb = QueryBuilder(limit=3).append(StructureData, project='id').order_by({StructureData:'id'})
-            qb2=QueryBuilder(**qb.queryhelp)
+            qb2 = QueryBuilder(**qb.queryhelp)
+
+            # The following is True if no change has been made to the database.
+            # Note that such a comparison can only be True if the order of results is enforced
+            qb.all() == qb2.all()
 
-        :return: a queryhelp dictionary
```

Comment:
still one json-compatible left here, I'm afraid ;-)

Reply:
otherwise looks good to merge for me

Reply:
done :) I'll squash and merge once the tests go through (and it is approved - again)